hazelcast/src/main/java/com/hazelcast/map/impl/operation/QueryOperation.java

```diff
@@ -171,9 +158,24 @@ protected void runParallelForPaging(Collection<Integer> initialPartitions) throw
         if (toMerge.size() > pagingPredicate.getPageSize()) {
             toMerge = toMerge.subList(0, pagingPredicate.getPageSize());
         }
-        for (QueryableEntry entry : toMerge) {
+        addQueryEntriesToResult(toMerge);
+    }
+
+    private static Collection<Collection<QueryableEntry>> getResult(List<Future<Collection<QueryableEntry>>> lsFutures) {
+        return returnWithDeadline(lsFutures, QUERY_EXECUTION_TIMEOUT_MINUTES, TimeUnit.MINUTES, FutureUtil.RETHROW_EVERYTHING);
+    }
+
+    private boolean addQueryEntriesToResult(Collection<QueryableEntry> queryableEntries) {
```

Comment:
The `boolean` return value doesn't make sense at the moment, but I will definitely use this in the next PR. That upcoming change will have a scenario for the `false` return statement.